mcstatus/__main__.py

```diff
@@ -1,75 +1,152 @@
 from __future__ import annotations
 
+import dns.resolver
+import sys
+import json
 import argparse
 import socket
-from json import dumps as json_dumps
+import warnings
+import dataclasses
+from typing import TYPE_CHECKING
 
-from mcstatus import JavaServer
+from mcstatus import JavaServer, BedrockServer
+from mcstatus.responses import JavaStatusResponse
+from mcstatus.motd import Motd
 
+if TYPE_CHECKING:
+    SupportedServers = JavaServer | BedrockServer
 
-def ping(server: JavaServer) -> None:
-    print(f"{server.ping()}ms")
 
+def _motd(motd: Motd) -> str:
+    """Formats MOTD for human-readable output, with leading line break
+    if multiline."""
+    s = motd.to_ansi()
+    return f"\n{s}" if "\n" in s else f" {s}"
 
-def status(server: JavaServer) -> None:
+
+def _kind(serv: SupportedServers) -> str:
+    if isinstance(serv, JavaServer):
+        return "Java"
+    elif isinstance(serv, BedrockServer):
+        return "Bedrock"
+    else:
+        raise ValueError(f"unsupported server for kind: {serv}")
+
+
+def _ping_with_fallback(server: SupportedServers) -> float:
+    # bedrock doesn't have ping method
+    if isinstance(server, BedrockServer):
+        return server.status().latency
+
+    # try faster ping packet first, falling back to status with a warning.
+    ping_exc = None
+    try:
+        return server.ping(tries=1)
+    except Exception as e:
+        ping_exc = e
+
+    latency = server.status().latency
+
+    address = f"{server.address.host}:{server.address.port}"
+    warnings.warn(
+        f"contacting {address} failed with a 'ping' packet but succeeded with a 'status' packet,\n"
+        f"  this is likely a bug in the server-side implementation.\n"
+        f'  (note: ping packet failed due to "{ping_exc}")\n'
+        f"  for more details, see: https://mcstatus.readthedocs.io/en/stable/pages/faq/\n",
+        stacklevel=1,
+    )
+
+    return latency
+
+
+def ping_cmd(server: SupportedServers) -> int:
+    print(_ping_with_fallback(server))
```

Comment:
Why not just include the `_ping_with_fallback` logic here directly? It's not like `__main__` is a part of mcstatus public API where this function would potentially be useful to any callers. It also doesn't seem like this function is re-used in the code.

If really we do want to go with this route, then `status_cmd`, `query_cmd` and `json_cmd` should also have their own functions, for consistency, no?.

Reply:
It was a rather OCD introduction to avoid multiple duplicated calls to print in the logic due to the fallback + Bedrock handling. Other commands have much simpler logic which do not require this.

Reply:
To respond more directly, I'm unconvinced of the need to change this. In fact, I can't even imagine what corresponding functions would do for the other commands. There is no reason to pursue "consistency" at all costs.

I'm any case, you are always able to correct this afterwards.